extension/observer/cfgardenobserver/config.go

```diff
@@ -0,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cfgardenobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver"
+
+import (
+	"time"
+)
+
+// Config defines configuration for CF Garden observer.
+type Config struct {
+	// The URL of the CF Garden api. Default is "unix:///var/vcap/data/garden/garden.sock"
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
Is the endpoint always a path, or can it be a URL with the http/s scheme? asking as this lib might come handy: https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confignet

Note this can be done later in a subsequent PR

Reply:
In CF is always a unix socket, but in terms of Garden it can be a http url. We will use this library! Thanks.